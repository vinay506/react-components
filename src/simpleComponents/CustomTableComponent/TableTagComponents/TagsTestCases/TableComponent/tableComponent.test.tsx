import React from 'react';
import { shallow } from 'enzyme';
import { CustomTbody, CustomThead, CustomTable, CustomTableContext} from '../../index';
import renderer from 'react-test-renderer'


describe('<Custom Table Tag >', function () {
    let props: any;
    let tree: any;
    let wrapper: any;
    let table: any;



    beforeEach(() => {
        props = {};
        table = <CustomTable {...props}>     <CustomThead>

        </CustomThead>
            <CustomTbody >
            </CustomTbody></CustomTable>
        wrapper = shallow(table);
    });

    // Snapshot test
    it('should render', () => {
        tree = renderer.create(table).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('checks with the className of Table', () => {
        const props = wrapper.find('table').props();
        expect(props.className).toEqual('mdc-data-table__table')
    })


    it('should have CustomTbody as child', () => {
        const customTbody = wrapper.find(CustomTbody);
        expect(customTbody).toHaveLength(1);
    })

    it('should have CustomThead as child', () => {
        const customThead = wrapper.find(CustomThead);
        expect(customThead.exists()).toBe(true);
    })

})


describe('<Custom Table Context >', function () {
    let props: any;
    let tree: any;
    let wrapper: any;
    beforeEach(() => {
        props = {};
        wrapper = shallow(<CustomTableContext {...props}><CustomTable {...props}>{props.children} </CustomTable></CustomTableContext>);
    });

    // Snapshot test
    it('should render', () => {
        tree = renderer.create(<CustomTableContext {...props}><CustomTable {...props}> {props.children}</CustomTable></CustomTableContext>).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('checks with the className of table context', () => {
        const props = wrapper.find('div').props();
        expect(props.className).toEqual('mdc-data-table')
    })
    it('should have customtable as child', () => {
        const customTable = wrapper.find(CustomTable);
        expect(customTable.exists()).toBe(true);
    })

})