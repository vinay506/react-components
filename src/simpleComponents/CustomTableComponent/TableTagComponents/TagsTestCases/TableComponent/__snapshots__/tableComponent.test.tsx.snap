// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Custom Table Context > should render 1`] = `
<div
  className="mdc-data-table"
>
  <table
    className="mdc-data-table__table"
  >
     
  </table>
</div>
`;

exports[`<Custom Table Tag > should render 1`] = `
<table
  className="mdc-data-table__table"
>
       
  <thead
    className="rmwc-data-table__head"
  />
  <tbody
    className="mdc-data-table__content"
  />
</table>
`;
